server/asset/dcr/dcr.go

```diff
@@ -399,7 +399,14 @@ func unconnectedDCR(logger dex.Logger) *Backend {
 // dcrd-registered handlers should perform any necessary type conversion and
 // then deposit the payload into the anyQ channel.
 func (dcr *Backend) Run(ctx context.Context) {
-	defer dcr.shutdown()
+	var wg sync.WaitGroup
+	wg.Add(1)
+	// Shut down the RPC client on ctx.Done().
+	go func() {
+		<-ctx.Done()
+		dcr.shutdown()
```

Comment:
Just to be clear, this forces a hung RPC call such as `dcr.node.GetBestBlockHash` to immediately return with an error, specifically `rpcclient.ErrClientShutdown`?

Reply:
Yes.
The error:
```
2020-05-05 10:02:04.651 [ERR] ASSET[dcr]: error retrieving best block: the client has been shutdown
```

rpcclient.ErrClientShutdown
https://github.com/decred/dcrd/blob/26c088aee05b47ee3a7a7cadfba8fe6834e471ee/rpcclient/infrastructure.go#L59-L63